src/fluree/db/json_ld/commit.cljc

```diff
@@ -350,6 +351,20 @@
           db**           (assoc-in db* [:ecount const/$_shard] @last-sid)]
       (add-commit-flakes-to-db db** flakes))))
 
+(defn link-context-to-commit
+  "Takes a commit with an embedded :context and pulls it out, saves it to
+  storage separately (content-addressed), and puts the address of that data
+  back into the commit under the :context key."
+  [{:keys [conn] :as ledger} commit]
+  (go-try
+    (let [context (get commit (keyword const/iri-default-context))
+          stringify? (-> context keys first keyword?) ; (too?) simple check if we need to stringify the keys before storing
```

Comment:
I think they should be JSON in the commit, so don't think this should ever be a keyword unless the JSON read function does recursive keywordizing, which maybe it does. Recursive keywordizing has given us many unintentional bugs, but if thats whats happening for now probably best to leave.